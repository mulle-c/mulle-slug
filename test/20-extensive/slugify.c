#include <mulle-slug/mulle-slug.h>

#include <stdio.h>


int   main( int argc, char *argv[])
{
   char   *s;
   char   *slug;
   int    c;

   mulle_buffer_do( buffer)
   {
      while( (c = getchar()) != EOF)
      {
         if( c == '\n')
         {
            s = mulle_buffer_extract_string( buffer);
            printf( "\"%s\" -> ", s);
            fflush( stdout);
            slug = mulle_utf8_slugify( s);
            printf( "\"%s\"\n", slug);
            mulle_free( slug);
            mulle_free( s);
            continue;
         }

         mulle_buffer_add_byte( buffer, c);
      }
   }

   return( 0);
}
