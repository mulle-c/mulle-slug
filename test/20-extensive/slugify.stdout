hashhash
dollar200
a-232-234
uZeurocruzeirofrench-francpound

https-maxmuster-geheim-www-example-com-8080-index-html-p1-A-p2-Bhashressource
mailto-max-example-org

lessa-href-hashfoo-bar-greater


ASCII-Table


2-3-4-5-6-7-30-40-50-60-70-80-90-100-110-120

0-0-P-p-0-2-less-F-P-Z-d-n-x
1-1-A-Q-a-q-1-3-G-Q-e-o-y
2-2-B-R-b-r-2-4-greater-H-R-f-p-z
3-hash-3-C-S-c-s-3-5-I-S-g-q
4-dollar-4-D-T-d-t-4-6-J-T-h-r
5-5-E-U-e-u-5-hash-7-A-K-U-i-s
6-6-F-V-f-v-6-dollar-8-B-L-V-j-t
7-7-G-W-g-w-7-9-C-M-W-a-k-u-DEL
8-8-H-X-h-x-8-0-D-N-X-b-l-v
9-9-I-Y-i-y-9-1-E-O-Y-c-m-w
A-J-Z-j-z
B-K-k
C-less-L-l
D-M-m
E-greater-N-n
F-O-o-DEL
